pyi.py

```diff
@@ -1319,6 +1344,28 @@ def _error_for_multiple_literals_in_union(
 
         self.error(first_union_member, Y030.format(suggestion=suggestion))
 
+    def _error_for_multiple_type_builtins_in_union(
+        self, first_union_member: ast.expr, analysis: UnionAnalysis, parent: str
+    ) -> None:
+        # Union is the explicit Union type, e.g. Union[type[str], type[int]]
+        if parent == "Union":
```

Comment:
I'm not a fan of the "magic strings" here. Could you use a boolean `is_pep_604_union` parameter instead of the `parent` parameter?

Reply:
Thanks for the suggestion! The main reason I didn't use a boolean was because I couldn't think of a descriptive name for it 😄